Readme suggestions (#84)
spacetelescope/dat_pyinthesky: jdat_notebooks/README.md

These notebooks can have many varied science cases, but follow a relatively
standard workflow:

1. Draft Stage
2. Baseline Stage
3. Notebook-driven development

> 4. Advanced Stage

Reviewer: ```suggestion
4. Advanced Integration Stage
```

I know we want to follow the DAWG wording, but I'm thinking we still want *something* that indicates this stage is really about "having the tools and the notebooks work together" instead of just "better" (which is what "advanced" sounds like to me).

I'm not necessarily attached to this *exact* wording but hopefully the idea gets across here.

Author: Sounds good to me!

thereby providing guidance for where specific development would simplify the workflow.

If a notebook is freshly created in this form, the author can follow the "Procedure to submit a notebook as a Pull Request"
(found at the end of this document), skipping the Draft Stage step.  

If the notebook was already created in the Draft Stage step and the "Procedure to submit a notebook as a Pull Request"
has already been followed, the author should just create a new branch to modify the existing code and then create 
a new Pull Request with the changes once they are ready.

In either case, the title (but not filename) of the notebook should begin with 

> "Baseline:" to indicate the notebook is in the Baseline Stage.

Reviewer: We haven't necessarily been doing this.  Do we want to stick with this, or should we change to the title just being the title and categorizing it using github or Jira?

(If we *do* still like this we should probably add it to the technical review to make sure the title is right...)

Author: I would like to have the title in the notebook (not in the file) to reflect the status, so "Baseline: blablabla...". Agree, we should add this to the technical review!

should be followed here.  The specific addition for the JWST Notebooks is that notebook data should be 
in the `DMD_Managed_Data/JWST/jwst-data_analysis_tools` Box folder (or subfolders thereof).  
If you do not have access to this box folder already, ask a Project Scientist and they should be able to get you added. 
Note that if a baseline notebook is using data that should not yet be public, the easiest choice is probably central store, 
but in that case it is critical that the notebook state prominently that it must be run inside the STScI network.
3. A notebook should state clearly what version of various dependencies were used to generate the notebook. 
These versions should be placed in a `requirements` file in the same directory as the notebook itself. An example of this file
is in the``example_notebook`` folder.
That will ensure reviewers/testers can be sure that if they encounter problems, it is not due to software version mis-matches.

The notebook will undergo a scientific and a technical review and it will be merged into the repository

> once the review comments have been addressed. This concludes the Baseline Stage.

Reviewer: ```suggestion
once the review comments have been addressed.  At this point the notebook has formally reached the Baseline Stage.
```

Maybe this is being overly-pedantic, but I think we want to call the notebook baseline when it's *done* with this process, rather than notebooks that are currently undergoing the process being called "baseline".

Author: Yes, I agree with you. The wording I used was misleading. It is clearer with your edit!

## Notebook-driven development

Along and after the Draft and Baseline stages, there is potential for considerable development 

> to be necessary.  A baseline notebook may contain a large number of areas where more development is desired in data 

Reviewer: ```suggestion
to be necessary in the libraries or other tools that the notebook uses.  A baseline notebook may contain a large number of areas where more development is desired in data 
```

Author: Ok

analysis tools, or it may only require a few minor adjustments (or none at all!).  This stage is therefore the most 
flexible and dependent on developer resources, etc.  In general the intent is for developers to be able to re-use 
bits of code from the notebook as tests for development, while occasionally (if necessary) asking the notebook 
author for guidance to ensure the implementation actually meets the notebook's needs.  There is not a formal 
process for this step, but it is intended that the JDAT planning process (currently on JIRA) keeps track of specific 
steps needed before a given notebook can proceed on to the next stage.

> ## Advanced Stage

Reviewer: ```suggestion
## Advanced Integration Stage
```
(or whatever is decided in my comment above)

ready to be called "Integrated".

Once the notebook authors (original author and developer/reviewer) have agreed it is ready, one of them follows 
the Pull Request workflow as described above, but with the notebook title now changed to be just 

> the title itself (no "Draft:" or Baseline:"). The Pull Request is then reviewed by one of the project scientists, and merged when 

Reviewer: I hadn't thought about this until now, but perhaps we want to replicate the sci/tech review process from the above at this time?  As it stands it's now just "the project scientist", which seems a bit of a bottleneck relative to the baseline review process.

(Of course the sci/tech review could *be* a project scientist, but we don't have to mandate it.)

Author: Right. Better say "by a developer and a scientist".
Thank you for catching these things!
